### PR TITLE
TSK-1258

### PR DESCRIPTION
**Pull request by nkolytschew**
Refactoring of spring versions in maven pom files

* move "spring-boot-dependencies" dependency to parent
* remove version from managed dependencies
* remove unused versions from parent

### CODE REVIEW

File: history/taskana-simplehistory-provider/pom.xml [hunk 1]
```diff
@@ -36,43 +35,36 @@
...
+			<artifactId>slf4j-simple</artifactId>
 			<scope>test</scope>
 		</dependency>
 		<dependency>
-			<groupId>org.apache.logging.log4j</groupId>
-			<artifactId>log4j-core</artifactId>
-			<version>${version.log4j}</version>
+			<groupId>org.slf4j</groupId>
+			<artifactId>jcl-over-slf4j</artifactId>
 			<scope>test</scope>
```
Comment (mustaphazorgati) [inline history/taskana-simplehistory-provider/pom.xml:48]: What is this dependency for?

File: lib/taskana-cdi/pom.xml [hunk 2]
```diff
@@ -25,29 +25,35 @@
...
+		<dependency>
+			<groupId>org.slf4j</groupId>
+			<artifactId>slf4j-simple</artifactId>
+			<scope>test</scope>
+		</dependency>
+		<dependency>
+			<groupId>org.slf4j</groupId>
+			<artifactId>jcl-over-slf4j</artifactId>
+			<scope>test</scope>
+		</dependency>
```
Comment (mustaphazorgati) [inline lib/taskana-cdi/pom.xml:37]: what is this dependency for?

File: lib/taskana-core/pom.xml [hunk 3]
```diff
@@ -79,15 +72,13 @@
...
+			<artifactId>slf4j-simple</artifactId>
 			<scope>test</scope>
 		</dependency>
 		<dependency>
-			<groupId>org.apache.logging.log4j</groupId>
-			<artifactId>log4j-core</artifactId>
-			<version>${version.log4j}</version>
+			<groupId>org.slf4j</groupId>
+			<artifactId>jcl-over-slf4j</artifactId>
 			<scope>test</scope>
```
Comment (mustaphazorgati) [inline lib/taskana-core/pom.xml:82]: what is this for?

File: lib/taskana-cdi/pom.xml (hunk 2 above)
Comment (nkolytschew) [reply lib/taskana-cdi/pom.xml]: fixed

**Comment by nkolytschew** [comment]
fixed review findings

File: lib/taskana-cdi/pom.xml (hunk 2 above)
Comment (mustaphazorgati) [reply lib/taskana-cdi/pom.xml]: I made a mistake yesterday. This dependency is necessary. Can you please re-add it?

File: lib/taskana-cdi/pom.xml [hunk 4]
```diff
@@ -25,29 +25,30 @@
 			<version>${project.version}</version>
 		</dependency>
 
+		<dependency>
+			<groupId>org.slf4j</groupId>
+			<artifactId>slf4j-simple</artifactId>
+			<scope>test</scope>
+		</dependency>
```
Comment (mustaphazorgati) [inline lib/taskana-cdi/pom.xml:32]: can you please move this down a line (under TEST dependencies) ?

File: lib/taskana-cdi/pom.xml (hunk 2 above)
Comment (nkolytschew) [reply lib/taskana-cdi/pom.xml]: reverted

File: lib/taskana-cdi/pom.xml (hunk 4 above)
Comment (nkolytschew) [reply lib/taskana-cdi/pom.xml:38]: fixed

**Pull request merged**